tests/unit/test_bruteforce.cpp

```diff
@@ -31,11 +31,104 @@ TEST_F(BruteForceTest, brute_force_vector_add_test) {
         a[i] = (float)i;
     }
     VecSimIndex_AddVector(index, (const void *)a, 1);
+
     ASSERT_EQ(VecSimIndex_IndexSize(index), 1);
+
     VecSimIndex_Free(index);
 }
 
-TEST_F(BruteForceTest, resizeIndex) {
+TEST_F(BruteForceTest, brute_force_vector_update_test) {
+    size_t dim = 4;
+    size_t n = 1;
+    size_t bs = DEFAULT_BLOCK_SIZE;
+    VecSimParams params{.algo = VecSimAlgo_BF,
+                        .bfParams = BFParams{.type = VecSimType_FLOAT32,
+                                             .dim = dim,
+                                             .metric = VecSimMetric_IP,
+                                             .initialCapacity = n}};
+    VecSimIndex *index = VecSimIndex_New(&params);
+    ASSERT_EQ(VecSimIndex_IndexSize(index), 0);
+
+    float a[dim];
+    for (size_t i = 0; i < dim; i++) {
+        a[i] = (float)1;
+    }
+    VecSimIndex_AddVector(index, (const void *)a, 1);
+    ASSERT_EQ(VecSimIndex_IndexSize(index), 1);
+
+    // Prepare new vector data.
+    for (size_t i = 0; i < dim; i++) {
+        a[i] = (float)2;
+    }
+    // Call addVEctor with the same id, different data.
+    VecSimIndex_AddVector(index, (const void *)a, 1);
+
+    // Index size shouldn't cahnge.
+    ASSERT_EQ(VecSimIndex_IndexSize(index), 1);
+
+    // id2label size should remain the same, although we seemingly tried to exceed
+    // initialcapacity.
+    ASSERT_EQ(reinterpret_cast<BruteForceIndex *>(index)->idToLabelMapping.size(), n);
+
+    // Check update.
+    VectorBlock *block = reinterpret_cast<BruteForceIndex *>(index)->getVectorVectorBlock(0);
+    float *vector_data = block->getVector(0);
+    for (size_t i = 0; i < dim; ++i) {
+        ASSERT_EQ(*vector_data, 2);
+        ++vector_data;
+    }
+
+    for (size_t i = 0; i < dim; i++) {
+        a[i] = (float)3;
+    }
+    // Add another vector and try to remove the first.
+    size_t last_vec_label = 2;
+    VecSimIndex_AddVector(index, (const void *)a, last_vec_label);
+    ASSERT_EQ(VecSimIndex_IndexSize(index), 2);
+    // Since we reached the limit of the initial capacity(1) id2label is
+    // increased and aligned with vector block size.
+    ASSERT_EQ(reinterpret_cast<BruteForceIndex *>(index)->idToLabelMapping.size(), bs);
+
+    size_t first_vec_label = 1;
+    size_t deleted_id = 0;
```

Comment:
isn't it this same logic as the swap test?
also, please rename `deleted_id` to `first_vec_new_id` or something similar. This is a bit confusing

Reply:
you are right. Swap test purpose is the to ensure the indexes of vector inside a block dont change upon deletion and this test also checks removing the last vector special case. ill move deletion of last vector to the delete swap test.